server.go

```diff
@@ -533,19 +532,24 @@ func (srv *Server) serveDTLSListener(l Listener) error {
 		rw, err := l.AcceptWithContext(ctx)
 		if err != nil {
 			wg.Wait()
```

Comment:
`wg` must be moved under `case ErrServerClosed` -> it used to wait for go-routines that served connections.